kernel/src/sched.rs

```diff
@@ -19,11 +27,49 @@ use crate::process::{self, Task};
 use crate::returncode::ReturnCode;
 use crate::syscall::{ContextSwitchReason, Syscall};
 
-/// The time a process is permitted to run before being pre-empted
-const KERNEL_TICK_DURATION_US: u32 = 10000;
 /// Skip re-scheduling a process if its quanta is nearly exhausted
 const MIN_QUANTA_THRESHOLD_US: u32 = 500;
 
+/// Trait which any scheduler must implement.
+pub trait Scheduler<C: Chip> {
+    /// Return the next process to run and the timeslice length for that process.
+    /// A timeslice of 0 means do preempt apps.
+    fn next(&self) -> (Option<AppId>, u32, bool);
```

Comment:
I still think it makes more sense semantically for timeslice to be an `Option` rather than "0 means infinity"

Reply:
Done